pilot/pkg/model/context.go

```diff
@@ -260,16 +260,7 @@ func GetProxyConfigNamespace(proxy *Proxy) string {
 		return configNamespace
 	}
 
-	// If its not present, fallback to parsing the namespace from the proxyID
-	// NOTE: This is a hack for Kubernetes only where we assume that the
-	// proxy ID is of the form name.namespace. Other platforms should
-	// set ISTIO_META_CONFIG_NAMESPACE in the sidecar bootstrap config.
-	parts := strings.Split(proxy.ID, ".")
```

Comment:
This is not right: we have id like `sidecar~172.17.0.13~istio-policy-84b5f87bfb-jq9xb.istio-system~istio-system.svc.cluster.local`.